.github/workflows/pr-check.yaml

```diff
@@ -319,13 +322,29 @@ jobs:
       - name: Execute pnpm
         run: pnpm install
 
-      - name: Run E2E smoke tests
-        run: pnpm test:e2e:smoke
+      - name: Build Podman Desktop for E2E tests Production Mode
+        if: ${{ matrix.mode == 'production' }}
+        env:
+          ELECTRON_ENABLE_INSPECT: true
+        run: |
+          pnpm compile:current --linux dir
+          path=$(realpath ./dist/linux-unpacked/podman-desktop)
+          echo "Podman Desktop built binary: $path"
+          echo "PODMAN_DESKTOP_BINARY_PATH=$path" >> $GITHUB_ENV
+
+      - name: Run Smoke E2E tests in ${{ matrix.mode }} mode
+        env:
+          PODMAN_DESKTOP_BINARY: ${{ env.PODMAN_DESKTOP_BINARY_PATH }}
+        run: |
+          if [ ${{ matrix.mode }} == 'production' ]; then
+            export PODMAN_DESKTOP_BINARY=${{ env.PODMAN_DESKTOP_BINARY_PATH }}
+          fi
+          pnpm test:e2e:smoke
```

Comment:
with all the env/export stuff, wondering then if it's not simpler, one section where we compile and run the tests for production and one section for development mode

```yaml
      - name: Run Smoke E2E tests in production mode
        if: ${{ matrix.mode == 'production' }}
        env:
          ELECTRON_ENABLE_INSPECT: true
        run: |
          pnpm compile:current --linux dir
          path=$(realpath ./dist/linux-unpacked/podman-desktop)
          echo "Podman Desktop built binary: $path"
          export PODMAN_DESKTOP_BINARY=$path
          pnpm test:e2e:smoke
          
     - name: Run Smoke E2E tests in development mode
        if: ${{ matrix.mode == 'development' }}
        run: |
          pnpm test:e2e:smoke          
```